homeassistant/components/brother/sensor.py

```diff
@@ -43,36 +52,41 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 class BrotherPrinterSensor(CoordinatorEntity, SensorEntity):
     """Define an Brother Printer sensor."""
 
-    def __init__(self, coordinator, kind, device_info):
+    def __init__(
+        self,
+        coordinator: BrotherDataUpdateCoordinator,
+        kind: str,
+        device_info: dict[str, Any],
+    ) -> None:
         """Initialize."""
         super().__init__(coordinator)
-        self._name = f"{coordinator.data.model} {SENSOR_TYPES[kind][ATTR_LABEL]}"
-        self._unique_id = f"{coordinator.data.serial.lower()}_{kind}"
+        self._name = f"{coordinator.data.model} {SENSOR_TYPES[kind][ATTR_LABEL]}"  # type: ignore[union-attr]
```

Comment:
Check that `coordinator.data` is not `None`.

Reply:
You mean something like this?
```python
if coordinator.data:
    self._name = f"{coordinator.data.model} {SENSOR_TYPES[kind][ATTR_LABEL]}"
    self._unique_id = f"{coordinator.data.serial.lower()}_{kind}"
```

Reply:
`coordinator.data` can be `None` only if first update failed.

You already checked in `async_setup_entry` that it's not `None` with `if sensor in coordinator.data:`.

So here I think just `assert coordinator.data is not None` in the beginning should be sufficient.

Reply:
Wow, it works. You've officialy become my typing guru :smile: